### PR TITLE
Make build work with Java17

### PR DESCRIPTION
**Pull request by cstamas**
In short: now build works on both, Java 11 and Java 17 (tested with 11 and 17 Temurin on Linux).

Also, build (plugins and tooling) is updated as well, dependencies are NOT touched except logging: they are set to slf4j 1.7.36 and logback 1.1.11. Reason is that (in tests) they were "crossed": slf4j 1.6 was used with 1.1.11 logback (while logback 1.1.x requires 1.7.x slf4j). So, dependencies can be set back (slf4j 1.6.x and logback 1.0.0) but then it needs to be consequently set (target AND test POM).

Changes:
* align POM with CONTRIBUTING (enforce Java11), up Maven requirement to 3.8.4+
* use javac `release=7` flag (instead of source/target+sniffer), remove animal sniffer
* fix enforcer override (was not working, as parent POM defined it in plugin not pluginMgmt section)
* move all plugin versions to top level POM
* update to latest plugins

Java 17 changes (commit 8fa9020c2faa5b7c1603e62e74668a64c4a69032):
* update felix and tycho versions to latest
* surefire: fixate junit47 provider, as presence of testng on classpath confuses surefire
* align PDE and POM declared logging lib versions (slf4j, logback), it caused clashes
* tests: exclude org.osgi:org.eclipse.osgi as it clashes (on test classpath) with felix
* tests: pass guiceVersion to UTs to be able to create "assumptions"
* ExampleTestCase `testContainerLookup` UT fails on Java 17 + Guice4 (did not look deeper), is disabled on that combo

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -289,22 +272,68 @@
...
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>build-helper-maven-plugin</artifactId>
+          <version>3.3.0</version>
         </plugin>
       </plugins>
     </pluginManagement>
+    <plugins>
+      <plugin>
+        <!-- Must have it here as pluginMgmt would NOT override since parent POM defined it here as well -->
```
Comment (kwin) [inline pom.xml:291]: I would rather recommend removing the no longer maintained parent oss-parent (https://central.sonatype.org/publish/publish-maven/#deprecated-oss-parent).

Comment (cstamas) [reply pom.xml:291]: Similar PR in mojos... https://github.com/eclipse/sisu.mojos/pull/7 maybe we need some sisu parent instead?

File: org.eclipse.sisu.inject/build.target [hunk 2]
```diff
@@ -17,19 +17,19 @@
 		<location type="Maven" includeSource="true">
 			<groupId>org.slf4j</groupId>
 			<artifactId>slf4j-api</artifactId>
-			<version>1.6.4</version>
```
Comment (kwin) [inline org.eclipse.sisu.inject/build.target:20]: If those dependency changes are not strictly necessary for Java 17 let us remove those. They are potentially backwards-incompatible.

File: pom.xml (hunk 1 above)
Comment (kwin) [reply pom.xml:291]: Let us track the introduction of a new parent pom.xml separately.

**Review by kwin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (kwin) [reply pom.xml:291]: Let us track this separately from this PR.

File: org.eclipse.sisu.inject/build.target (hunk 2 above)
Comment (cstamas) [reply org.eclipse.sisu.inject/build.target:20]: If I undo these changes, I get errors in tests (LogsTest.testProductionLogging:38 » IllegalState org.slf4j.LoggerFactory could not be successfully initialized. See also http://www.slf4j.org/codes.html#unsuccessfulInit)

**Comment by cstamas** [comment]
Frankly, I'd drop this PR and rather continue with https://github.com/eclipse/sisu.inject/pull/71 that incorporates this one as well....

**Comment by cstamas** [comment]
Superseded by https://github.com/eclipse/sisu.inject/pull/71

**Pull request closed without merging**